### PR TITLE
US123931 Added header/footer dialog button to accordion.

### PR DESCRIPTION
**Pull request by ian-wallace**
https://rally1.rallydev.com/#/16349626795d/dashboard?detail=%2Fuserstory%2F482797593636

### CODE REVIEW

File: components/d2l-activity-editor/d2l-activity-quiz-editor/d2l-activity-quiz-manage-header-footer-container.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+				.d2l-activity-quiz-manage-header-footer-container-dialog-summary {
+					margin: 0.5rem 0;
+				}
+				.d2l-activity-quiz-manage-header-footer-container-dialog-summary d2l-icon {
+					margin-right: 0.3rem;
+				}
+				:host([dir="rtl"]) .d2l-activity-quiz-manage-header-footer-container-dialog-summary d2l-icon {
+					margin-left: 0.3rem;
+					margin-right: 0;
+				}
```
Comment (mdulat) [inline components/d2l-activity-editor/d2l-activity-quiz-editor/d2l-activity-quiz-manage-header-footer-container.js:33]: It doesn't look like any of the styles specified are relevant here. I.e. none of the specifiers exist.

File: components/d2l-activity-editor/d2l-activity-quiz-editor/d2l-activity-quiz-manage-header-footer-container.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+	}
+
+	constructor() {
+		super(store);
+	}
+
+	firstUpdated() {
+		super.firstUpdated();
+		this.serverErrorTerm = this.localize('quizHeaderFooterServerError');
+		this.validationErrorTerm = this.localize('quizHeaderFooterValidationError');
```
Comment (mdulat) [inline components/d2l-activity-editor/d2l-activity-quiz-editor/d2l-activity-quiz-manage-header-footer-container.js:45]: I don't see any error handling or validation for this story, maybe double-check with Joseph, but if there is no requirement then all the error stuff can be removed from here.

File: components/d2l-activity-editor/d2l-activity-quiz-editor/d2l-activity-quiz-manage-header-footer-container.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+			<d2l-button-subtle text=${this.localize('manageHeaderFooter')} @click="${this.openDialog}" h-align="text"></d2l-button-subtle>
+		`;
+	}
+
+	_renderQuizHeaderFooterEditor() {
+		const entity = store.get(this.dialogHref);
+		if (!entity) return html``;
+
+		const {
+			timingHref
```
Comment (mdulat) [inline components/d2l-activity-editor/d2l-activity-quiz-editor/d2l-activity-quiz-manage-header-footer-container.js:73]: I don't think you need this, probably a copy/paste from the timing stuff

File: components/d2l-activity-editor/d2l-activity-quiz-editor/d2l-activity-quiz-manage-header-footer-container.js [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+		} = entity || {};
+
+		return html`
+			<d2l-activity-quiz-manage-header-footer-editor
+				href="${timingHref}"
```
Comment (mdulat) [inline components/d2l-activity-editor/d2l-activity-quiz-editor/d2l-activity-quiz-manage-header-footer-container.js:78]: Probably the href would just be this.href (whatever will return the header value).

File: components/d2l-activity-editor/d2l-activity-quiz-editor/lang/en.js [hunk 5]
```diff
@@ -114,5 +114,9 @@ export default {
 	"quizTimingServerError": "Something went wrong. Please try again.", // Timing save server error alert message
 	"quizTimingSummary": "{timingType} ({numMinutes, plural, =1 {1 minute} other {{numMinutes} minutes}})", // Timing type followed by (x minute) or (x minutes). e.g. Recommended time limit (120 minutes)
 	"ipRestrictionsDuplicateError": "Duplicate IP range start address. Each IP range start value must be unique.", // Error for duplicate IP
-	"ipRestrictionsRangeError": "Invalid IP address range provided. Please ensure ranges are correctly formatted." // Error for invalid IP ranges
+	"ipRestrictionsRangeError": "Invalid IP address range provided. Please ensure ranges are correctly formatted.", // Error for invalid IP ranges
+	"subHdrHeaderFooter": "Header and Footer", // Header and Footer accordion header
+	"manageHeaderFooter": "Manage Header and Footer", // Label for button to launch Header and Footer dialog
+	"quizHeaderFooterServerError": "Oops server error", // replace with real error string
+	"quizHeaderFooterValidationError": "Oops header / footer validation error" // replace with real string
```
Comment (mdulat) [inline components/d2l-activity-editor/d2l-activity-quiz-editor/lang/en.js:121]: I wouldn't add the error lang terms, they may get accidentally translated, and probably we don't need any for header.

File: components/d2l-activity-editor/d2l-activity-quiz-editor/d2l-activity-quiz-manage-header-footer-container.js (hunk 1 above)
Comment (ian-wallace) [reply components/d2l-activity-editor/d2l-activity-quiz-editor/d2l-activity-quiz-manage-header-footer-container.js:33]: I will remove all of the styles.  I was thinking they would be relevant for the dialog, but it looks like not.  When the dialog gets in I'll add what's needed then.

**Review by mdulat — APPROVED** [review]

File: components/d2l-activity-editor/d2l-activity-quiz-editor/d2l-activity-quiz-manage-header-footer-container.js (hunk 1 above)
Comment (ian-wallace) [reply components/d2l-activity-editor/d2l-activity-quiz-editor/d2l-activity-quiz-manage-header-footer-container.js]: I broke it - I need to put some of these back I think.

**Pull request closed without merging**